dash/include/dash/halo/Halo.h

```diff
@@ -294,8 +476,21 @@ class HaloSpec {
   constexpr HaloSpec(const Specs_t& specs) : _specs(specs) {}
 
   template <typename StencilSpecT>
-  HaloSpec(const StencilSpecT& stencil_specs) {
-    for(const auto& stencil : stencil_specs.specs()) {
+  HaloSpec(const StencilSpecT& stencil_spec) {
+    for(const auto& stencil : stencil_spec.specs()) {
+      auto stencil_combination = stencil;
+
+      set_region_spec(stencil_combination);
+      while(next_region(stencil, stencil_combination)) {
+        set_region_spec(stencil_combination);
+      }
+    }
+  }
+
+  template <typename StencilSpecT, typename... Args>
+  HaloSpec(const StencilSpecT& stencil_spec, const Args&... stencil_specs)
+  : HaloSpec(stencil_specs...) {
```

Comment:
What happens to the leading `stencil_spec` argument?

Reply:
it was used in the for loop, now it is used in the function read_stencil_points